src/ClientWidgetApi.ts

```diff
@@ -844,12 +828,20 @@ export class ClientWidgetApi extends EventEmitter {
             );
         } catch (e) {
             console.error("error while downloading a file", e);
-            this.transport.reply<IWidgetApiErrorResponseData>(request, {
-                error: { message: "Unexpected error while downloading a file" },
-            });
+            this.handleDriverError(e, request, "Unexpected error while downloading a file");
         }
     }
 
+    private handleDriverError(e: unknown, request: IWidgetApiRequest, message: string) {
+        const data = this.driver.processError(e);
```

Comment:
I'm not a fan of making driver implementations responsible for serializing errors.  I'd much rather have it be done by something akin to a method on the error object itself, but that's not simple because errors are of an unknown type here.  For now, the current solution works, but I'm open to suggestions.